### PR TITLE
Adds Shuttle 667 as a buyable shuttle for dispelling a cult rune

### PR DESCRIPTION
**Pull request by Mothblocks**
## About The Pull Request
Adds Shuttle 667 (the "Nar Nar" shuttle) as a purchasable shuttle. You can buy it once a cult rune is dispelled through a null rod.

Also adds two stasis beds in the medbay part.

image

## Why It's Good For The Game
Adds more novelty to shuttle purchasing. Plus, who doesn't want to see a cult attempt to hijack the emergency shuttle to use the summon rune on it?

## Changelog
:cl:
add: You can now buy the Shuttle 667 when a cult rune is dispelled with a null rod.
/:cl:

### CODE REVIEW

**Comment by itseasytosee** [comment]
The shuttle looks ok, but the requirements would need to be a bit steeper for sure.

**Comment by Jared-Fogle** [comment]
Why? What do you recommend?

File: code/modules/antagonists/cult/runes.dm [hunk 1]
```diff
@@ -64,6 +64,7 @@ Runes can either be invoked by one's self or with many different cultists. Each
 	else if(istype(I, /obj/item/nullrod))
 		user.say("BEGONE FOUL MAGIKS!!", forced = "nullrod")
 		to_chat(user, "<span class='danger'>You disrupt the magic of [src] with [I].</span>")
+		SSshuttle.shuttle_purchase_requirements_met |= SHUTTLE_UNLOCK_NARNAR
```
Comment (Rohesie) [inline code/modules/antagonists/cult/runes.dm:67]:
```suggestion
		SSshuttle.shuttle_purchase_requirements_met[SHUTTLE_UNLOCK_NARNAR] = TRUE
```

File: code/datums/shuttles.dm [hunk 2]
```diff
@@ -334,6 +332,10 @@
 	description = "Looks like this shuttle may have wandered into the darkness between the stars on route to the station. Let's not think too hard about where all the bodies came from."
 	admin_notes = "Contains real cult ruins, mob eyeballs, and inactive constructs. Cult mobs will automatically be sentienced by fun balloon. \
 	Cloning pods in 'medbay' area are showcases and nonfunctional."
+	credit_cost = 6667
+
+/datum/map_template/shuttle/emergency/narnar/prerequisites_met()
+	return SHUTTLE_UNLOCK_NARNAR in SSshuttle.shuttle_purchase_requirements_met
```
Comment (Rohesie) [inline code/datums/shuttles.dm:338]:
```suggestion
	return SSshuttle.shuttle_purchase_requirements_met[SHUTTLE_UNLOCK_NARNAR]
```

**Comment by ATH1909** [comment]
Can the constructs on the shuttle participate in the activation of the rune?

~~Imagine if some cultist intentionally clears a rune with a null rod, calls this specific shuttle, and then pops out a couple of shades on the rune and activates it.~~

**Comment by ATH1909** [comment]
Oh, wait, you can't activate the summon rune if it's not in one of the designated summon locations (I just checked).

Sad!

**Comment by Sparkezel** [comment]
Oh, thats trurly sad I believe, is it supposed to be something like one time cult support drop-ship?

**Comment by FloranOtten** [comment]
If I can remember correctly, don't those cult barriers disappear after a few seconds? Last time I spawned it, I had to replace those and repressurize the shuttle.

**Comment by itseasytosee** [comment]
First off this shuttle needs some TLC, Its old and hasn't to be updated in forever. Second I don't really think you can go too high on the requirement here, if no other ideas are brought up you could make it an option in dynamic mode when a cult is defeated. Another option could be if the cult leader is killed.

**Comment by ATH1909** [comment]
> If I can remember correctly, don't those cult barriers disappear after a few seconds? Last time I spawned it, I had to replace those and repressurize the shuttle.

Yeah, I think I noticed that while codediving a while back. IIRC, multiple ruins use the kinds of cult forcefields that cult juggernauts make, which disappear on their own after a bit. Coincidentally, since they're not subtypes of wizard forcefields, people with antimagic (and even the juggernaut who made them) can't walk through juggernaut forcefields.

File: _maps/shuttles/emergency_narnar.dmm [hunk 3]
```diff
@@ -6,7 +6,9 @@
 /turf/closed/wall/mineral/cult,
 /area/shuttle/escape)
 "c" = (
-/obj/effect/forcefield/cult,
+/obj/effect/forcefield/cult{
+	timeleft = 0
```
Comment (Rohesie) [inline _maps/shuttles/emergency_narnar.dmm:10]: Could you make a variant object instead?

**Comment by Jared-Fogle** [comment]
> If I can remember correctly, don't those cult barriers disappear after a few seconds? Last time I spawned it, I had to replace those and repressurize the shuttle.

They did, yeah. I had to fix it.

**Comment by Barhandar** [comment]
Emergency shuttle should be an always valid narnar summoning location

**Comment by Tlaltecuhtli** [comment]
looks sad

**Comment by optimumtact** [comment]
I don't want to have to ban new shuttles but it's getting to the point where I might have to given how many exist for no real reason

**Comment by Jared-Fogle** [comment]
The reason for their existence is just diversity. They have historically been extremely low maintenance.

**Comment by coiax** [comment]
I made this shuttle.

It makes me so happy to see it again. ❤️

**Pull request merged**